stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -138,15 +138,23 @@ internal class PaymentOptionsActivity : BasePaymentSheetActivity<PaymentOptionRe
                 }
             }
         }
+    }
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
+    private fun setupAddButton(addButton: AddButton) {
+        addButton.completedAnimation.observe(this) { completedState ->
+            completedState?.paymentSelection?.let { paymentSelection ->
+                animateOut(
+                    PaymentOptionResult.Succeeded(paymentSelection)
+                )
+            }
+        }
```

Comment:
Does this handle the case where we don't animate?

Reply:
Now it does :D